test/libsolidity/SolidityNameAndTypeResolution.cpp

```diff
@@ -102,7 +102,13 @@ parseAnalyseAndReturnError(string const& _source, bool _reportWarnings = false,
 		if (success)
 			if (!StaticAnalyzer(errors).analyze(*sourceUnit))
 				success = false;
-		if (errors.size() > 1 && !_allowMultipleErrors)
+		int errorCount = 0;
+		for (auto const& currentError: errors)
+		{
+			if (currentError->type() != Error::Type::Warning)
```

Comment:
If possible, I would prefer to keep this as strict as it is now. How many tests would need changes?